Spanner/src/Result.php

```diff
@@ -228,13 +223,19 @@ public function rows($format = self::RETURN_ASSOCIATIVE)
 
                 // retry without resume token when results have not yielded
                 $shouldRetry = !$isResultsYielded || $hasResumeToken;
-                $generator->next();
-                $valid = $generator->valid();
+                $this->generator->next();
+                $valid = $this->generator->valid();
             } catch (ServiceException $ex) {
+                $backoff = new ExponentialBackoff($this->retries, function ($ex) {
+                    if ($ex instanceof ServiceException) {
+                        return $ex->getCode() === Grpc\STATUS_UNAVAILABLE;
+                    }
+                    return false;
+                });
```

Comment:
big nit, but we can move this into the `if` statement since we don't need it unless that condition is true.

Reply:
Okay, as the `try` specifically `catch` `ServiceException`, we can do that.

Reply:
We don't even need the `if` condition I think.

Reply:
No, I mean to move the backoff object